sympy/sets/sets.py

```diff
@@ -686,6 +716,36 @@ def is_iterable(self):
         """
         return all(set.is_iterable for set in self.sets)
 
+    @property
+    def is_FiniteSet(self):
```

Comment:
I would expect a capitalised name like this to mean `isinstance(self, FiniteSet)`. Maybe `is_finite_set` is better.